src/main/java/opengl/fbo/FBO.java

```diff
@@ -0,0 +1,308 @@
+        idFrameBuffer = (int) intHandle.get(frameBufferIds, 0);
+
+        if(debug)
+            System.out.println("FBO: Got FB ID : " + idFrameBuffer);
+
+        // Bind frame buffer
+        gl.glBindFramebufferEXT(gl.GL_FRAMEBUFFER_EXT(), idFrameBuffer);
+
+        //Attach 2D texture to this FBO
+        gl.glFramebufferTexture2DEXT(gl.GL_FRAMEBUFFER_EXT(), gl.GL_COLOR_ATTACHMENT0_EXT(), gl.GL_TEXTURE_2D(), idTexture, 0);
+
+        //-------------------------
+        // Generate RENDER buffer
+
+        renderBufferIds = MemorySegment.allocateNative(4, newImplicitScope());
+        gl.glGenRenderbuffersEXT(1, renderBufferIds);
+        idRenderBuffer = (int) intHandle.get(renderBufferIds, 0);
+
+        if(debug)
+            System.out.println("FBO: Got RenderBuffer ID : " + idRenderBuffer);
+
+        // Bind render buffer
+        gl.glBindRenderbufferEXT(gl.GL_RENDERBUFFER_EXT(), idRenderBuffer);
+        gl.glRenderbufferStorageEXT(gl.GL_RENDERBUFFER_EXT(), gl.GL_DEPTH_COMPONENT24(), width, height);
+
+        //-------------------------
+        //Attach depth buffer to FBO
+
+        gl.glFramebufferRenderbufferEXT(gl.GL_FRAMEBUFFER_EXT(), gl.GL_DEPTH_ATTACHMENT_EXT(), gl.GL_RENDERBUFFER_EXT(), idRenderBuffer);
+
+        //-------------------------
+        //Does the GPU support current FBO configuration?
+
+        int status = gl.glCheckFramebufferStatusEXT(gl.GL_FRAMEBUFFER_EXT());
+
+        if(status!=gl.GL_FRAMEBUFFER_COMPLETE()){
+            throw new RuntimeException("Incomplete framebuffer, not supporting current FBO config : " + status + " != GL_FRAMEBUFFER_COMPLETE (" + gl.GL_FRAMEBUFFER_COMPLETE() + ")");
+        }
+
+        //-------------------------
+        //and now you can render to GL_TEXTURE_2D
+
+        gl.glBindFramebufferEXT(gl.GL_FRAMEBUFFER_EXT(), idFrameBuffer);
+        gl.glClearColor(0.0f, 0.0f, 0.0f, 0.0f);
+        gl.glClearDepth(1.0f);
+        gl.glClear(gl.GL_COLOR_BUFFER_BIT() | gl.GL_DEPTH_BUFFER_BIT());
+
+        // Mark as prepared
+        prepared = true;
+    }
+
+    /**
+     * Release resources held by this FBO utility.
+     *
+     * @param gl
+     */
+    public void release(GL gl){
+        //Delete resources
+        gl.glDeleteTextures(1, textureBufferIds);
+        gl.glDeleteRenderbuffersEXT(1, renderBufferIds);
+        //Bind 0, which means render to back buffer, as a result, fb is unbound
+        gl.glBindFramebufferEXT(gl.GL_FRAMEBUFFER_EXT(), 0);
+        gl.glDeleteFramebuffersEXT(1, frameBufferIds);
+
+        // FIXME : Not mapped exception is not relevant
+        // FIXME : See if later versions of Panama do not throw exception
+        /*textureBufferIds.unload();
+        renderBufferIds.unload();
+        frameBufferIds.unload();
+        pixelBuffer.unload();*/
+
+        prepared = false;
+    }
+
+    public BufferedImage getImage(GL gl) {
+
+        // Initialize buffers if they are not ready or if their expected size changed
+        if(!prepared)
```

Comment:
Most probably no sense : it is required to invoke the GL scene BETWEEN preparation and image generation.